operators/pkg/controller/common/overrides/env.go

```diff
@@ -0,0 +1,66 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package overrides
+
+import (
+	corev1 "k8s.io/api/core/v1"
+)
+
+// EnvBuilder helps with building a list of environment variables while
+// dealing with name conflicts.
+type EnvBuilder struct {
+	byName  map[string]corev1.EnvVar
```

Comment:
Coudl this just be a `map[string]struct{}` to get the set semantics without duplicating the env var data? You are never actually using the values. 

Alternatively `map[string]bool` would also simplify this idiom a bit:
```golang
if _, exists := e.byName[v.Name]; exists {
			// a variable with the same name already exists, keep the existing one
			continue
		}
```
to 
```golang
if e.byName[v.Name] {
	// a variable with the same name already exists, keep the existing one
	continue
}
```

Reply:
Yeah you're totally right!